src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -520,9 +573,11 @@ private static boolean validateHostname(String host) {
         if ((ch < 'a' || ch > 'z') && (ch < '0' || ch > '9') && ch != '-') {
```

Comment:
Some small helper functions can make this more readable like `if (!isAlpha(ch) && !isDigit(cg) && ch != '-')`

Reply:
Fixed.